src/components/ProductsSelect/index.tsx

```diff
@@ -67,7 +67,14 @@ export const ProductsSelect: React.FC<{
       allowClear
       style={{ width: '100%', minWidth: '150px' }}
       value={currProducts}
-      onChange={onChange}
+      onChange={(changedValue) => {
+        if (!changedValue) {
+          setCurrProducts([]);
+        }
+        if (onChange) {
+          onChange(changedValue);
+        }
```

Comment:
not very reactive - but i'm might not seeing whole context 

shouldn't be `useEffect` set to observable `CurrProducts` then fire `onChange` ?

Reply:
We need clear state where we handling selected products. onChange is optional prop. I can't do this in useEffect